nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -237,31 +223,48 @@ bool CollisionMonitor::getParameters(
 
     nav2_util::declare_parameter_if_not_declared(
       node, "base_frame_id", rclcpp::ParameterValue("base_footprint"));
-    std::string base_frame_id = get_parameter("base_frame_id").as_string();
+    base_frame_id_ = get_parameter("base_frame_id").as_string();
     nav2_util::declare_parameter_if_not_declared(
       node, "transform_tolerance", rclcpp::ParameterValue(0.1));
-    tf2::Duration transform_tolerance =
+    transform_tolerance_ =
       tf2::durationFromSec(get_parameter("transform_tolerance").as_double());
     nav2_util::declare_parameter_if_not_declared(
       node, "data_timeout", rclcpp::ParameterValue(2.0));
     data_timeout_ =
       tf2::durationFromSec(get_parameter("data_timeout").as_double());
-    nav2_util::declare_parameter_if_not_declared(
-      node, "simulation_time_step", rclcpp::ParameterValue(0.02));
-    double simulation_time_step = get_parameter("simulation_time_step").as_double();
 
     nav2_util::declare_parameter_if_not_declared(
-      node, "release_acceleration", rclcpp::ParameterValue(0.1));
-    release_acceleration_ = get_parameter("release_acceleration").as_double();
+      node, "approach", rclcpp::ParameterValue(false));
+    approach_ = get_parameter("approach").as_bool();
+  } catch (const std::exception & ex) {
+    RCLCPP_ERROR(get_logger(), "Error while getting parameters: %s", ex.what());
+    return false;
+  }
 
-    // Leave these parameters to be not initialized:
-    // the will cause an error if it will not set
-    nav2_util::declare_parameter_if_not_declared(
-      node, "polygons", rclcpp::PARAMETER_STRING_ARRAY);
-    nav2_util::declare_parameter_if_not_declared(
-      node, "observation_sources", rclcpp::PARAMETER_STRING_ARRAY);
+  if (!configurePolygons()) {
+    return false;
+  }
+
+  if (approach_) {
+    if (!configureFootprint(footprint_topic)) {
```

Comment:
Since usually robot has only one footprint (maybe I am false, but I did not met more than one footprint per a robot before), I've decided to have one footprint for Collision Monitor, too. That in my opinion is more clear to developer using CM as ROS-node. However, indeed, this only complicated the code.

> That's how it was before and was perfect

If you think so, I am switching back to old model, where all polygons were STOP/SLOWDOWN/APPROACH.

Reply:
But you could potentially have multiple approach patterns on different footprint topics. I don't think that's a _likely_ situation, but it would be annoying if someone came up with a good point and we'd have to re-introduce multiple polyons for approach models again (given that it would have worked before). 

A situation I could think of is a mobile manipulator where the robot base has a footprint but if the arm drops below a certain dimension it also publishes a footprint.

Reply:
Okay, I've switched back to old multi-footprint usage model.